src/aggregator/aggregator/capture/aggregator.go

```diff
@@ -131,6 +132,26 @@ func (agg *aggregator) AddForwarded(
 	return nil
 }
 
+func (agg *aggregator) AddPassThrough(
+	metric aggregated.Metric,
+	metadata metadata.TimedMetadata,
+) error {
+	// Clone the metric and timed metadata to ensure it cannot be mutated externally.
+	metric = cloneTimedMetric(metric)
+	metadata = cloneTimedMetadata(metadata)
+
+	agg.Lock()
+	defer agg.Unlock()
```

Comment:
i would say avoid the defer and just unlock below

Reply:
Using defer is consistent with other AddXxxx functions.

Reply:
Yep so this aggregator implementation (i.e. the one in `src/aggregator/aggregator/capture`) is only used in tests, so not as critical. Fine to leave as is here but wouldn't do that in the general impl (the one used in prod).